src/log-common.js

```diff
@@ -9,15 +9,133 @@
+ */
+const suppressProgress = (fields) => {
+  // eslint-disable-next-line no-underscore-dangle,no-console
+  if (fields.progress && console._stderr.isTTY) {
+    return undefined;
+  }
+  // eslint-disable-next-line no-param-reassign
+  delete fields.progress;
+  return fields;
+};
+/**
+ * remove progress field
+ */
+const filterProgress = (fields) => {
+  // eslint-disable-next-line no-param-reassign
+  delete fields.progress;
+  return fields;
+};
+
+const messageFormatConsole = (fields) => {
+  // eslint-disable-next-line
+  const {level, timestamp, message, category = 'cli', ...rest} = fields;
+
+  const fullMsg = Object.keys(rest).length === 0 ? message : [...message, ' ', rest];
+  const ser = serializeMessage(fullMsg, { colors: false });
+
+  let lvl = level.toLowerCase();
+  if (colors[level]) {
+    lvl = chalk[colors[level]](lvl);
+  }
+  if (category === 'cli') {
+    if (level === 'info') {
+      return `${ser}`;
+    } else {
+      return `${lvl}: ${ser}`;
+    }
+  } else {
+    return chalk`{grey [${category}]} ${lvl}: ${ser}`;
+  }
+};
 
-function makeLogger({ logLevel = 'info', logFile = ['-'] } = {}) {
-  return Logger.getLogger({
-    category: 'cli',
-    logFile,
-    level: logLevel,
+// set the default logger, in case code uses root logger directly.
+rootLogger.loggers.get('default').formatter = messageFormatConsole;
+
+// module global loggers by category
+const loggersByCategory = new Map();
+
+function makeLogger(config) {
+  let categ;
+  if (typeof config === 'string') {
+    categ = config;
+  } else {
+    categ = (config && config.category) || 'cli';
+  }
+
+  if (loggersByCategory.has(categ)) {
+    return loggersByCategory.get(categ);
+  }
+
+  // setup helix logger
+  const level = (config && config.level) || (this && this.level) || 'info';
+  const logsDir = path.normalize((config && config.logsDir) || 'logs');
+  const logFiles = config && Array.isArray(config.logFile)
+    ? config.logFile
+    : ['-', (config && config.logFile) || path.join(logsDir, `${categ}-server.log`)];
+
+  const loggers = new Map();
+  logFiles.forEach((logFile) => {
+    const name = loggers.has('default') ? logFile : 'default';
+    if (logFile === '-') {
+      loggers.set(name, new ConsoleLogger({
+        filter: categ === 'cli' ? suppressProgress : filterProgress,
+        level,
+        formatter: messageFormatConsole,
+      }));
+    } else {
+      fs.ensureDirSync(path.dirname(logFile));
+      loggers.set(name, new FileLogger(logFile, {
+        level: 'debug',
+        formatter: /\.json/.test(logFile) ? messageFormatJsonString : messageFormatTechnical,
+      }));
+    }
+  });
+
+  // create simple interface
+  const log = new SimpleInterface({
+    level,
+    defaultFields: {
+      category: categ,
+    },
+    logger: new MultiLogger(loggers),
   });
+
+  // create proxy to handle `getLogger` method
+  const handler = {
+    get(target, prop) {
+      if (prop === 'getLogger') {
```

Comment:
Is this even used anywhere? I can't find any invocations?

Reply:
yes. can be removed now, I think.